extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioHandlerFactory.java

```diff
@@ -35,13 +35,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
 
     @Override
     protected ThingHandler createHandler(Thing thing) {
-
-        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
-        if (thingTypeUID.equals(THING_TYPE_RADIO)) {
-            return new FSInternetRadioHandler(thing);
-        }
-
-        return null;
+        return new FSInternetRadioHandler(thing);
```

Comment:
Well, the fact that the framework currently only calls it for thing types that were returned on supportedThingTypes might not be a solid fact - at least I didn't find any such assertions in the JavaDocs or in the code of the BaseThingHandlerFactory (where this method is defined and used).
Looking from a pure Java perspective, this is a method of an interface and as the implementor, I should make the implementation robust within the range that the interface and its JavaDoc allows. I therefore would not approve this change. Besides this, it actually also makes it more difficult for future contributors to figure out what needs to be changed if a new supported thing type is added to the binding - with the old code, this is very obvious.